coalib/testing/LocalBearTestHelper.py

```diff
@@ -26,20 +26,45 @@ def execute_bear(bear, *args, **kwargs):
     return list(bear_output_generator)
 
 
+def get_local_bear_run_settings_dict(local_bear, settings_dict):
```

Comment:
This should not be necessary here.

It should be a method of `Bear` or `FunctionMetadata`, if something similar & sensible doesnt already exist.

Also this algorithm should be recursive (c.f. https://github.com/coala/coala/pull/3663)